# Homework solution - 04 Model Deployment

In this notebook are the answers of the homework of the module 04 of the course (Model Deployment).

In [7]:
# Built-in imports
import os
import json

# External imports
import pandas as pd

# Own imports
from scripts import get_path_dir as gpd

DATA_DIR = gpd.get_desired_folder_path("data")

## Q1. Notebook

The mean of the predicted duration for the `fhv_tripdata_2021-02.parquet` dataset is `16.191691681964873`.

Refer to [ride-duration-prediction](ride-duration-prediction.ipynb).

## Q2. Preparing the output

The size of the output parquet file is approximately of `39 MB`.

Refer to [ride-duration-prediction](ride-duration-prediction.ipynb).

## Q3. Creating the scoring script

To convert the jupyter notebook into a python script the `jupyter nbconvert --to script <notebook_file.ipynb> --output <python_file.py>` command has to be used.

Refer to [ride-duration-prediction](ride-duration-prediction.ipynb)

## Q4. Virtual environment

In [4]:
with open("../Pipfile.lock", "r") as read_file:
    scores = json.load(read_file)

first_hash = scores["default"]["scikit-learn"]["hashes"][0]
first_hash

'sha256:08ef968f6b72033c16c479c966bf37ccd49b06ea91b765e1cc27afefe723920b'

The first hash for the Scikit-Learn dependency is `sha256:08ef968f6b72033c16c479c966bf37ccd49b06ea91b765e1cc27afefe723920b`.

## Q5. Parametrize the script

The mean of the predicted duration for the `fhv_tripdata_2021-03.parquet` dataset is `16.298821598819046`.

Refer to [ride-duration](../scripts/ride-duration.py.py).

## Q6. Docker container

In [8]:
!docker run --rm mlops/zoomcamp/ride_prediction:1.0 -y 2021 -m 4

Making the predictions ... 
The mean predicted duration is 9.967573179784523


The mean of the predicted duration for the `fhv_tripdata_2021-04.parquet` dataset is `9.967573179784523`.